src/pyk/proof/equality.py

```diff
@@ -218,3 +312,41 @@ def advance_proof(self, kcfg_explore: KCFGExplore) -> None:
 
         _LOGGER.info(f'EqualityProof finished {self.proof.id}: {self.proof.status}')
         self.proof.write_proof()
+
+
+class RefutationProver:
+    proof: RefutationProof
+
+    def __init__(self, proof: RefutationProof) -> None:
+        self.proof = proof
+
+    def advance_proof(self, kcfg_explore: KCFGExplore) -> None:
+        _LOGGER.info(f'Attempting RefutationProof {self.proof.id}')
+
+        if self.proof.status is not ProofStatus.PENDING:
+            _LOGGER.info(f'RefutationProof finished {self.proof.id}: {self.proof.status}')
+            return
+
+        antecedent_kast = mlAnd(self.proof.pre_constraints)
+        consequent_kast = mlEqualsFalse(self.proof.last_constraint)
+
+        antecedent_simplified_kast, _ = kcfg_explore.kast_simplify(antecedent_kast)
+        consequent_simplified_kast, _ = kcfg_explore.kast_simplify(consequent_kast)
+        self.proof.set_simplified_constraints(consequent_simplified_kast)
+        _LOGGER.info(f'Simplified constraints: {kcfg_explore.kprint.pretty_print(consequent_simplified_kast)}')
+
+        if is_top(consequent_simplified_kast):
+            _LOGGER.warning(f'Consequent of implication (proof equality) simplifies to #Top {self.proof.id}')
+            self.proof.set_csubst(CSubst(Subst({}), ()))
```

Comment:
```suggestion
        if is_top(consequent_simplified_kast):
            _LOGGER.warning(f'Consequent of implication (proof equality) simplifies to #Top {self.proof.id}')
            self.proof.set_csubst(CSubst(Subst({}), ()))
        if is_bottom(antecedent_simplified_kast):
            _LOGGER.warning(f'Antecedent of implication (proof equality) simplifies to #Top {self.proof.id}')
            self.proof.set_csubst(CSubst(Subst({}), ()))
```

Now, it looks quite a bit exactly the same as `EqualityProver`? Perhaps we should be inheriting from that class now?

Reply:
We can worry about that later.